pkg/new-ui/v1beta1/authzn.go

```diff
@@ -0,0 +1,116 @@
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/kubeflow/katib/pkg/util/v1beta1/env"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// ENV variables
+var (
+	USER_HEADER  = env.GetEnvOrDefault("USERID_HEADER", "kubeflow-userid")
+	USER_PREFIX  = env.GetEnvOrDefault("USERID_PREFIX", ":")
+	DISABLE_AUTH = env.GetEnvOrDefault("APP_DISABLE_AUTH", "false") == "true"
+	BACKEND_MODE = env.GetEnvOrDefault("BACKEND_MODE", "prod")
+)
+
+func GetUsername(r *http.Request) (string, error) {
+	var username string
+	if DISABLE_AUTH {
+		log.Printf("APP_DISABLE_AUTH set to True. Skipping authorization check")
+		return "", nil
+	}
+
+	if r.Header.Get(USER_HEADER) == "" {
+		return "", errors.New("User header not present!")
+	}
+
+	user := r.Header.Get(USER_HEADER)
+	username = strings.Replace(user, USER_PREFIX, "", 1)
+
+	return username, nil
+}
+
+// Function for constructing SubjectAccessReviews (SAR) objects
+func CreateSAR(user string, verb string, namespace string, group string,
+	version string, resource string, subresource string, name string) *v1.SubjectAccessReview {
```

Comment:
Should we remove duplicated variable types like `func CreateSAR(foo, bar string)`?

Reply:
done